tests/utilities/test_meta.py

```diff
@@ -68,3 +70,17 @@ def test_init_meta_context():
 
     m = nn.Linear(in_features=1, out_features=1)
     assert m.weight.device.type == "cpu"
+
+
+@RunIf(min_torch="1.10.0", standalone=True)
+def test_fit_with_init_meta_context():
+    """Test fitting a model initialized on meta device."""
+    with init_meta_context():
+        model = BoringModel()
+    trainer = Trainer(
+        fast_dev_run=True,
+        enable_progress_bar=False,
+        enable_model_summary=False,
+        enable_checkpointing=False,
+    )
+    trainer.fit(model)
```

Comment:
Fitting is not necessary. Let's instead target a unit test here (directly on the function / decorator)